shared/util/pictures.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+import {apiserverGetRpc} from '../constants/types/flow-types'
+import {throttle} from 'lodash'
+
+type Info = {
+  url: ?string,
+  callbacks: Array<(username: string, url: ?string) => void>,
+  done: boolean,
+  error: boolean,
+}
+
+// Done
+const _usernameToURL: {[key: string]: ?Info} = {}
+// Not done
+const _pendingUsernameToURL: {[key: string]: ?Info} = {}
+
+const _getUserImages = throttle(() => {
+  const usersToResolve = Object.keys(_pendingUsernameToURL)
+  if (!usersToResolve.length) {
+    return
+  }
+
+  // Move pending to non-pending state
+  usersToResolve.forEach(username => {
+    const info: ?Info = _pendingUsernameToURL[username]
+    _usernameToURL[username] = info
```

Comment:
i don't think so. it should never overwrite while moving over here. This is some of the complexity thats involved in having 2 lists